core/runner/src/test/java/com/cognifide/aet/runner/processing/data/SuiteDataServiceTest.java

```diff
@@ -0,0 +1,121 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.cognifide.aet.runner.processing.data;
+
+import static org.mockito.Matchers.anyObject;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.cognifide.aet.communication.api.metadata.Suite;
+import com.cognifide.aet.communication.api.metadata.SuiteBuilder;
+import com.cognifide.aet.communication.api.metadata.ValidatorException;
+import com.cognifide.aet.vs.MetadataDAO;
+import com.cognifide.aet.vs.StorageException;
+import java.util.concurrent.ThreadLocalRandom;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class SuiteDataServiceTest {
+
+  private static final String CHECKSUM_01 = "checksum01";
+
+  private static final String COMPANY = "company";
+
+  private static final String PROJECT = "project";
+
+  private static final String SUITE_NAME = "suite-name";
+
+  private static final String patternCorrelationId = "company-project-suite-name-123456789";
+
+  private static final String ID_LAST_VERSION_SUITE = "idLastVersionSuite";
+
+  private final MetadataDAO metadataDAO = mock(MetadataDAO.class);
+
+  private Suite currentRunWithChecksum;
+
+  private Suite currentRunWithoutCorrelationId;
+
+  private Suite lastVersionSuite;
+
+  private Suite suiteHashcode;
+
+
+  private SuiteDataService service;
+
+  @Before
+  public void setUp() {
+
+    currentRunWithChecksum = new SuiteBuilder()
+        .setCorrelationId(patternCorrelationId)
+        .setCompany(COMPANY).setProject(PROJECT)
+        .setName(SUITE_NAME)
+        .setPatternCorrelationId("id2")
+        .setProjectHashCode(CHECKSUM_01)
+        .createSuite();
+
+    currentRunWithoutCorrelationId = new SuiteBuilder()
+        .setCorrelationId(patternCorrelationId)
+        .setCompany(COMPANY).setProject(PROJECT)
+        .setName(SUITE_NAME)
+        .setPatternCorrelationId(null)
+        .createSuite();
+
+    lastVersionSuite = new SuiteBuilder().setCorrelationId(patternCorrelationId)
+        .setCompany(COMPANY).setProject(PROJECT)
+        .setName(SUITE_NAME)
+        .setPatternCorrelationId(ID_LAST_VERSION_SUITE)
+        .setProjectHashCode(CHECKSUM_01)
+        .createSuite();
+    lastVersionSuite.setVersion(generateRandomPositiveLong());
+
+    suiteHashcode = new SuiteBuilder().setCorrelationId(patternCorrelationId)
+        .setCompany(COMPANY).setProject(PROJECT)
+        .setName(SUITE_NAME)
+        .setPatternCorrelationId("id")
+        .setProjectHashCode(CHECKSUM_01)
+        .createSuite();
+    service = new SuiteDataService(metadataDAO);//todo OSGI mock
```

Comment:
I have a problem with OSGI mock. At the moment mock is inject by constructor

Reply:
You can delete this line, annotate class with 
```
@RunWith(MockitoJUnitRunner.class)
```
and annotate service object field with
```
@InjectMocks
private SuiteDataService service;
```
and it will work.